nexus/test-utils/src/lib.rs

```diff
@@ -1142,33 +1122,40 @@ impl<'a, N: NexusServer> ControlPlaneTestContextBuilder<'a, N> {
             schema_version: 2,
         };
 
-        client
-            .write_network_bootstore_config(&early_network_config)
-            .await
-            .expect("Failed to write early networking config to bootstore");
+        macro_rules! from_clone {
+            ($source: expr) => {
+                $source.clone().into_iter().map(From::from).collect()
+            };
+        }
 
-        for sled_agent in self.sled_agents.iter().skip(1) {
+        let generation = Generation::from_u32(2);
+        let zones =
+            once(from_clone!(self.blueprint_zones)).chain(repeat(vec![]));
```

Comment:
Is this constructing a structure with non-empty zones for the first sled agent and no zones for the others?

Reply:
Yes, it's the same pattern used in the blueprint builder's `maybe_zones`. Commented in 4c1f2ab01.